include/cantera/base/FactoryBase.h

```diff
@@ -74,28 +74,56 @@ class Factory : public FactoryBase {
 
     //! Create an object using the object construction function corresponding to
     //! "name" and the provided constructor arguments
-    T* create(std::string name, Args... args) {
-        try {
-            return m_creators.at(name)(args...);
-        } catch (std::out_of_range&) {
-            if (m_synonyms.find(name) != m_synonyms.end()) {
-                return m_creators.at(m_synonyms.at(name))(args...);
-            } else if (m_deprecated_names.find(name) != m_deprecated_names.end()) {
-                warn_deprecated(name,
-                    fmt::format("Use '{}' instead.", m_deprecated_names.at(name)));
-                return m_creators.at(m_deprecated_names.at(name))(args...);
-            } else {
-                throw CanteraError("Factory::create", "No such type: '{}'", name);
-            }
-        }
+    T* create(const std::string& name, Args... args) {
+        return m_creators.at(canonicalize(name))(args...);
     }
 
     //! Register a new object construction function
     void reg(const std::string& name, std::function<T*(Args...)> f) {
         m_creators[name] = f;
     }
 
+    //! Add an alias for an existing registered type
+    void addAlias(const std::string& original, const std::string& alias) {
+        if (!m_creators.count(original)) {
+            throw CanteraError("Factory::addAlias",
+                "Name '{}' not registered", original);
+        }
+        m_synonyms[alias] = original;
+    }
+
+    //! Get the canonical name registered for a type
+    std::string canonicalize(const std::string& name) {
```

Comment:
Would a better name for this function be `getCanonicalName()`? I guess Wikipedia [says](https://en.wikipedia.org/wiki/Canonicalization) that this name is the correct one, but I sure can't pronounce it 😄

Reply:
Maybe, but it conflicts with the convention in our C++ code that `getFoo` names are mainly used for methods where something is being "returned" in a reference/pointer argument to the function. I guess "normalize" would be a simpler word, even if this particular usage is just as esoteric.

Reply:
OK, seems fine then.